src/app/shared/layout/header.component.ts

```diff
@@ -37,6 +46,32 @@ export class HeaderComponent implements OnInit {
     }
   }
 
+  reload() {
+    this.isReloadBtnDisabled = true;
+
+    this.githubEventService.getLatestChangeEvent().subscribe((eventResponse) => {
+        // Will only allow page to reload if the latest modify time is different
+        // from last modified, meaning that some changes to the repo has occured.
+        if (eventResponse['created_at'] !== this.githubEventService.getLastModifiedTime() ||
+        eventResponse['issue']['updated_at'] !== this.githubEventService.getLastModifiedCommentTime()) {
+          this.issueService.reloadAllIssues().subscribe(
```

Comment:
To reduce nesting of subscription, you can use pipes and flatmaps.

Reply:
I have made the necessary changes